plutus-benchmark/bitwise/bench/Main.hs

```diff
@@ -1,26 +1,33 @@
+-- editorconfig-checker-disable-file
+{-# LANGUAGE DataKinds         #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE TemplateHaskell   #-}
+
 module Main (main) where
 
-{-
 import Criterion.Main (bench, defaultMain)
 import PlutusBenchmark.Common (benchProgramCek, mkMostRecentEvalCtx)
+import PlutusBenchmark.Ed25519 (checkValid)
 import PlutusBenchmark.NQueens (nqueens)
 import PlutusTx.Code (CompiledCode, getPlcNoAnn)
 import PlutusTx.TH (compile)
--}
-
-main :: IO ()
-main = print "Pending"
-
-{- Currently not able to run, due to problems with writeBits compiling under PlutusTx
 
 main :: IO ()
 main = defaultMain [
-  bench "8-queens" . benchProgramCek mkMostRecentEvalCtx . getPlcNoAnn $ nqueensCompiled
+  bench "8-queens" . benchProgramCek mkMostRecentEvalCtx . getPlcNoAnn $ nqueensCompiled,
+  bench "Ed25519" . benchProgramCek mkMostRecentEvalCtx . getPlcNoAnn $ ed25519Compiled
   ]
 
+
 -- Helpers
 
+ed25519Compiled :: CompiledCode Bool
```

Comment:
It's better to make `CompiledCode (Signature -> Pk -> Bool)`, then use `unsafeApplyCode` to obtain the `CompiledCode Bool`. This ensures that nothing is evaluated at compile time. [This module](https://github.com/IntersectMBO/plutus/blob/master/plutus-tx-plugin/test/AsData/Budget/Spec.hs) has some examples.

Reply:
I guess this also applies to the 8 queens benchmark?

Reply:
Yes that's correct